load_test/amoc_scenarios/xmpp_amqp.erl

```diff
@@ -302,6 +310,18 @@ amqp_do(State) ->
     end,
     amqp_do(NewState).
 
+-spec process_presence_message(#'amqp_msg'{}) -> ok.
+process_presence_message(#'amqp_msg'{payload = P}) ->
+    lager:debug("Got presence message ~p", [P]),
+    ok.
+
+-spec process_chat_message(#'amqp_msg'{}) -> ok.
+process_chat_message(#'amqp_msg'{payload = P}) ->
+    lager:debug("Got chat message ~p", [P]),
```

Comment:
Maybe we should first update metrics to make measurement as accurate as we can? I also think that it could be better to log a decoded message.

Reply:
- You are right, its better to make measurement as fast as we can.
- I think it is better to log RAW message. It may be malformed.

Reply:
The only way that a message gets malformed and decoded process fails is when you change the Mongoose source code. In such a situation tests wouldn't pass so I'm not sure if the case you described fits here. IMO logging decoded message would increase log readability.